perf: temp_scaler, bp_balance, load_least,

youmeiyoumaotai opened the pull request:
ensure_scaler, ensure_scheduler, lass, pass,
rotate.
and Three types of load plots

ActivePeter commented on scripts/draw_script/analyse_experiment_data.py:
```diff
@@ -0,0 +1,131 @@
+# 自动运行同路径下的run_different_req_freq.py
+import os
+CUR_FPATH = os.path.abspath(__file__)
```
对应的说明文档在顶上注释链接下

YouMeiYouMaoTai replied:
这两个绘图文件是草稿，忘记删除了

ActivePeter commented on serverless_sim/src/scale/num/ensure_scaler.rs:
```diff
@@ -22,18 +23,40 @@ impl EnsureScaleNum {
-        }
-        else {
-            cur_container_cnt + sqrt_container_cnt
+        if need_to_schedule || cur_container_cnt == 0 {
```
当前容器数为0，不需要调度的情况，为什么要走这个逻辑，走了这个逻辑就到==0的case，返回1了

YouMeiYouMaoTai replied:
一开始没加这一块，实验跑出来发现冷启动延迟很大，文章中控制缩容的时候还关注了slo，但是在系统实现这两个算法的时候时间比较紧，没有实现这个slo来控制缩容，所以给这块补偿了这个逻辑来尽量接近原来的冷启动情况。

ActivePeter commented on serverless_sim/src/scale/num/lass.rs:
```diff
@@ -56,49 +74,24 @@ impl ScaleNum for LassScaleNum {
+
+                log::info!("fnid:{}, frame:{}, recent_speed:{}", fnid, env.core().current_frame(), recent_speed);
+
+                if recent_speed < 0.00001 && need_to_schedule{
```
这里和下面的1000取值都是怎么来的

ActivePeter commented on serverless_sim/src/scale/num/temp_scaler.rs:
这个是自己论文的话建议飞书里建个文档仔细的介绍下，连接到主页

ActivePeter commented on serverless_sim/src/sche/ensure_scheduler.rs:
```diff
@@ -88,7 +83,7 @@ impl Scheduler for EnsureScheduler {
                 req,
                 env,
-                schedule_helper::CollectTaskConfig::PreAllSched,
+                schedule_helper::CollectTaskConfig::All,
```
这一块确认清楚，尤其是scaler那里和这并不一样，最好引用一下文档，在文档里解释下原文实现和这个的映射

ActivePeter commented on serverless_sim/src/sche/pass.rs:
pass的实现怎么是随机，我记得不是静态贪婪嘛

ActivePeter reviewed (changes requested):
Request changes

YouMeiYouMaoTai commented on serverless_sim/src/scale/num/ensure_scaler.rs:
```diff
@@ -22,18 +23,40 @@ impl EnsureScaleNum {
+            let schedule_able_fns = schedule_helper::collect_task_to_sche(
+                req,
+                env,
+                schedule_helper::CollectTaskConfig::PreAllSched,
```
当时想测试这三种有什么性能上的区别，测完没改回来

youmeiyoumaotai committed:
perf: temp_scaler, bp_balance, load_least,
ensure_scaler, ensure_scheduler, lass, pass,
rotate.
and Three types of load plots

youmeiyoumaotai committed:
Merge branch 'main' into DELL

youmeiyoumaotai committed:
fix：修正 .gitingore 中的 cache 文件夹

youmeiyoumaotai committed:
feat: update scaling strategies and experiment configs

youmeiyoumaotai committed:
Merge branch 'DELL' of https://github.com/YouMeiYouMaoTai/serverless_sim into DELL

youmeiyoumaotai committed:
feat：新增kneeScale复现、gvgc实现

youmeiyoumaotai committed:
复现 bcws 算法

youmeiyoumaotai committed:
复现：第四章节新机制rbuc、faascache、flame、s-cache
